backend/app/api/routes/students.py

```
@@ -58,9 +61,21 @@ async def upload_image_url(
 
     student_user_id = ObjectId(current_user["id"])
-
+    
+    # 1. Read image bytes
+    image_bytes = await file.read()
+    
+    # 2. Generate face embeddings
+    try:
+        embedding = get_face_embedding(image_bytes)
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)